### PR TITLE
JP-2918: Fix bad resampled frames for NRS MOS

### PR DESCRIPTION
**Pull request by hbushouse**
Resolves [JP-2918](https://jira.stsci.edu/browse/JP-2918)

This PR fixes the calculation of resampled output frames for NIRSpec slit data (either fixed-slit or MOS) by making the algorithm less sensitive to the presence of negative spectral traces in the input images (which occurs when nodding is used to do background subtraction).

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by hbushouse** [comment]
This initial commit is a crude first attempt that appears to do a decent job of fixing the problem. It simply ignores all pixels with values < 0 when computing the weighted means used to find the target location and hence makes it less susceptible to negative spectral traces and even just random large negative values.

**Review by mcara — COMMENTED** [review]
Seems reasonable, except, I am not sure about the need of the quite non-informative logging messages introduced in this PR.

**Comment by hbushouse** [comment]
> Seems reasonable, except, I am not sure about the need of the quite non-informative logging messages introduced in this PR.

Looks like our comments just crossed paths!  The extra logging messages were just for my debugging work and will be removed again once we're OK with the fix.

File: jwst/resample/resample_spec.py [hunk 1]
```diff
@@ -99,6 +99,9 @@ def build_nirspec_output_wcs(self, refmodel=None):
         else:
             refmodel = self.input_models[0]
 
+        refmodel_data = refmodel.data.copy()
+        refmodel_data[refmodel_data < 0.] = np.NaN
```
Comment (mcara) [inline jwst/resample/resample_spec.py:103]: One possible way to improve the logic would be to find the minimum value in data and subtract it from data:
```suggestion
        min_refmodel_data = np.nanmin(refmodel_data)
        refmodel_data -= min_refmodel_data
```

Comment (mcara) [reply jwst/resample/resample_spec.py:103]: Another approach would be to add back subtracted sky if that value was recorded in the metadata.

**Comment by mcara** [comment]
@hbushouse What actually prompted this PR? What was happening when data had negative values? `wmean_s` is essentially a "center of mass" estimate. IRAF's `imexamine` actually used to take data and subtract mean value from data before computing CofM, see, e.g., https://github.com/mcara/imexamine-centroid/blob/f696adae0aaf15eb4de4c3b6474903f018390b6f/centroid.py#L258-L282

File: jwst/resample/resample_spec.py (hunk 1 above)
Comment (hbushouse) [reply jwst/resample/resample_spec.py:103]: Unfortunately this is not a case of having a constant background subtracted. It's images with the spectra in different places that've been subtracted from one another, so you're left with both positive and negative traces of what is essentially the same spectrum.

**Comment by hbushouse** [comment]
> @hbushouse What actually prompted this PR? What was happening when data had negative values?

See the last comment in JP-2918. Briefly, sometimes it was getting values of `wmean_s` that were like -3510. When that value was used in the wcs transform to go to the sky frame, it was estimating source RA/Dec values that were way off from the real values. For example, it would compute RA~105, Dec~13, when the real RA/Dec were +155.9 and -57.8, respectively. That in turn caused it to use really big offsets ("ts") that got added to the y slit min/max. So the whole output frame was totally wrong.

File: jwst/resample/resample_spec.py (hunk 1 above)
Comment (hbushouse) [reply jwst/resample/resample_spec.py]: I implemented your suggested change to renormalize to the min value. Seems to work well without explicitly excluding some of the data (like my initial approach did).

**Comment by hbushouse** [comment]
CI errors are unrelated problems with CodeCov of some kind.

**Review by mcara — APPROVED** [review]
LGTM

**Pull request merged**